clientd3d/music.c

```diff
@@ -145,23 +154,66 @@ DWORD OpenMidiFile(const char *lpszMIDIFileName)
       debug(("Unable to get current directory!\n"));
 
    sprintf(filename, "%s%s", current_dir, lpszMIDIFileName);
-   debug(("music filename = %s\n", filename));
-
+   debug(("music filename = %s \n", filename));
+   // Is it a background music or gameplay element music file?
+   if (loadWhat == LOAD_MUSIC)
```

Comment:
A safer way is to use a switch statement here, with a default, with a default case that prints an error.  That way if the caller does pass in an illegal value, it will be noticed.

It would in general also make it easier to add more cases in the future (though that seems unlikely for this particular enum).

Reply:
This is done.  It reads better too.